modules/platforms/dotnet/Apache.Ignite.Core.Tests/Client/Services/ServiceAwarenessTest.cs

```diff
@@ -0,0 +1,518 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Core.Tests.Client.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Apache.Ignite.Core.Client;
+    using Apache.Ignite.Core.Client.Services;
+    using Apache.Ignite.Core.Cluster;
+    using Apache.Ignite.Core.Log;
+    using Apache.Ignite.Core.Services;
+    using Apache.Ignite.Core.Tests.Client.Cache;
+    using Apache.Ignite.Core.Tests.Services;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Service awareness tests.
+    /// </summary>
+    public class ServicesAwarenessTest
```

Comment:
This class reimplements a lot of logic from `ClientTestBase` - is that intentional?

Reply:
Yes. I saw it. Not so much shared logic. Just cluster and logs management. I would need to fix the base classs too. But ok, implemented relying on ClientTestBase